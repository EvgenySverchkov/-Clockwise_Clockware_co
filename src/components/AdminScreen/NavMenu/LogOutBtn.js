import React from "react";
import { useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import { toogleAuth } from "../../../store/adminPanel/services/actions";

const LogOutBtn = () => {
  const dispatch = useDispatch();

  function handler() {
    sessionStorage.removeItem("token");
    dispatch(toogleAuth(false));
  }
  return (
    <div className="mr-1">
      <Link
        to="/admin"
        onClick={handler}
        className="btn btn-secondary"
        aria-haspopup="true"
        aria-expanded="false"
      >
        LogOut
      </Link>
    </div>
  );
};

export default LogOutBtn;
