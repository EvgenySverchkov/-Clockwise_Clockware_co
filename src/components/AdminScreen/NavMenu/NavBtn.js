import React from "react";

export default ({ title, children }) => (
  <div className="dropdown mr-1">
    <button
      className="btn btn-secondary dropdown-toggle"
      href="#"
      id="dropdownMenuLink"
      data-toggle="dropdown"
      aria-haspopup="true"
      aria-expanded="false"
    >
      {title}
    </button>
    <div className="dropdown-menu" aria-labelledby="dropdownMenuLink">
      {children}
    </div>
  </div>
);
