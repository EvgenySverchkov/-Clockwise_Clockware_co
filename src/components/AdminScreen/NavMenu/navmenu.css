.nav-menu-wrapp {
  text-align: center;
}
