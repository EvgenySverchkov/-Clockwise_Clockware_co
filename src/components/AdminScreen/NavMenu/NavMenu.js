import React from "react";
import { Link } from "react-router-dom";
import "./navmenu.css";

function NavMenu() {
  return (
    <div className="nav-menu-wrapp mb-3 mb-sm-5">
      <div className="btn-group nav-menu">
        <div className="dropdown mr-1">
          <button
            className="btn btn-secondary dropdown-toggle"
            href="#"
            id="dropdownMenuLink"
            data-toggle="dropdown"
            aria-haspopup="true"
            aria-expanded="false"
          >
            Orders
          </button>
          <div className="dropdown-menu" aria-labelledby="dropdownMenuLink">
            <Link to="/admin/ordersList" className="dropdown-item">
              Show orders
            </Link>
          </div>
        </div>
        <div className="dropdown mr-1">
          <button
            className="btn btn-secondary dropdown-toggle"
            type="button"
            id="dropdownMenuButton"
            data-toggle="dropdown"
            aria-haspopup="true"
            aria-expanded="false"
          >
            Masters
          </button>
          <div className="dropdown-menu" aria-labelledby="dropdownMenuLink">
            <Link to="/admin/mastersList" className="dropdown-item">
              Show masters
            </Link>
            <Link to="/admin/addMasterForm" className="dropdown-item">
              Add master
            </Link>
          </div>
        </div>
        <div className="dropdown">
          <button
            className="btn btn-secondary dropdown-toggle"
            type="button"
            id="dropdownMenuButton"
            data-toggle="dropdown"
            aria-haspopup="true"
            aria-expanded="false"
          >
            Towns
          </button>
          <div
            className="dropdown-menu dropdown-menu-right"
            aria-labelledby="dropdownMenuLink"
          >
            <Link to="/admin/townsList" className="dropdown-item">
              Show towns
            </Link>
            <Link to="/admin/addTownForms" className="dropdown-item">
              Add town
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
}

export default NavMenu;
