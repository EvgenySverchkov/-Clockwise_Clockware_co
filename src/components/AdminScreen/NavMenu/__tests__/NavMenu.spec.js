import React from "react";
import {create} from "react-test-renderer";
import {MemoryRouter} from "react-router-dom"
import { Provider } from "react-redux";
import consigMockStore from "redux-mock-store";

import NavMenu from "../NavMenu";

const mockStore = consigMockStore();

it("Test of <NavMenu /> on admin side", ()=>{
    const store = mockStore();
    const component = create(
        <Provider store={store}>
            <MemoryRouter initialEntries={["/"]}>
                <NavMenu />
            </MemoryRouter>
        </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
});