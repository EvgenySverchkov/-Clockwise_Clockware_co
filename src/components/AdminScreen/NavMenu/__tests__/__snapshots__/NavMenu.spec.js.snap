// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <NavMenu /> on admin side 1`] = `
<div
  className="nav-menu-wrapp mb-3 mb-sm-5"
>
  <div
    className="btn-group nav-menu"
  >
    <div
      className="dropdown mr-1"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        className="btn btn-secondary dropdown-toggle"
        data-toggle="dropdown"
        href="#"
        id="dropdownMenuLink"
      >
        Orders
      </button>
      <div
        aria-labelledby="dropdownMenuLink"
        className="dropdown-menu"
      >
        <a
          className="dropdown-item"
          href="/admin/ordersList"
          onClick={[Function]}
        >
          Show orders
        </a>
        <a
          className="dropdown-item"
          href="/admin/addOrderForm"
          onClick={[Function]}
        >
          Add order
        </a>
      </div>
    </div>
    <div
      className="dropdown mr-1"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        className="btn btn-secondary dropdown-toggle"
        data-toggle="dropdown"
        href="#"
        id="dropdownMenuLink"
      >
        Masters
      </button>
      <div
        aria-labelledby="dropdownMenuLink"
        className="dropdown-menu"
      >
        <a
          className="dropdown-item"
          href="/admin/mastersList"
          onClick={[Function]}
        >
          Show masters
        </a>
        <a
          className="dropdown-item"
          href="/admin/addMasterForm"
          onClick={[Function]}
        >
          Add master
        </a>
      </div>
    </div>
    <div
      className="dropdown mr-1"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        className="btn btn-secondary dropdown-toggle"
        data-toggle="dropdown"
        href="#"
        id="dropdownMenuLink"
      >
        Towns
      </button>
      <div
        aria-labelledby="dropdownMenuLink"
        className="dropdown-menu"
      >
        <a
          className="dropdown-item"
          href="/admin/townsList"
          onClick={[Function]}
        >
          Show towns
        </a>
        <a
          className="dropdown-item"
          href="/admin/addTownForms"
          onClick={[Function]}
        >
          Add town
        </a>
      </div>
    </div>
    <div
      className="mr-1"
    >
      <a
        aria-expanded="false"
        aria-haspopup="true"
        className="btn btn-danger"
        href="/admin"
        onClick={[Function]}
      >
        LogOut
      </a>
    </div>
  </div>
</div>
`;
